eng/pipelines/templates/jobs/ci.yml

```diff
@@ -226,18 +228,51 @@ jobs:
         parameters:
           BaselineFilePath: $(Build.sourcesdirectory)/eng/dotnet.gdnbaselines
           ServiceDirectory: ${{ parameters.ServiceDirectory }}
-
   - template: /eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml
     parameters:
+      GenerateJobName: generate_target_service_test_matrix
       JobTemplatePath: /eng/pipelines/templates/jobs/ci.tests.yml
       MatrixConfigs: ${{ parameters.MatrixConfigs }}
       MatrixFilters: ${{ parameters.MatrixFilters }}
       MatrixReplace: ${{ parameters.MatrixReplace }}
       CloudConfig:
-        Cloud: Public
+        Cloud: public
       AdditionalParameters:
         SDKType: ${{ parameters.SDKType }}
-        ServiceToTest: ${{ coalesce(parameters.ServiceToTest, parameters.ServiceDirectory) }}
         ServiceDirectory: ${{ parameters.ServiceDirectory }}
         TestSetupSteps: ${{ parameters.TestSetupSteps }}
-        TestTimeoutInMinutes: ${{ parameters.TestTimeoutInMinutes }}
+        TestTimeoutInMinutes: ${{ parameters.TestTimeoutInMinutes }} 
+  - ${{ if ne(parameters.TestDependsOnDependency, 'not-specified') }}:
+    - template: /eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml
+      parameters:
+        GenerateJobName: generate_target_dependencies_test_matrix 
+        JobTemplatePath: /eng/pipelines/templates/jobs/ci.tests.yml
+        MatrixConfigs: 
+          - Name: NET_ci_test_base
+            # The path here should refer to dependency.tests.yml parameter ProjectFilesOutputFolder 
+            Path: $(Build.ArtifactStagingDirectory)/platform-matrix.json
+            Selection: sparse
+            GenerateVMJobs: true
+        MatrixFilters: ${{ parameters.MatrixFilters }}
+        MatrixReplace: ${{ parameters.MatrixReplace }}
+        CloudConfig:
+          Cloud: public
+        SparseCheckoutPaths: 
+          - "/*"
```

Comment:
Why do we need to check out everything, rather than just the service directory and eng?

Reply:
For this case the prestep needs access to all the project files and anything needed to call the target (i.e. *.target, *.props, etc) so I suggested we just checkout everything except the test recordings. We definitely need more than the one service directory.

Reply:
For the one running on service directory, we sparse-checkout on service directory only.
For the one running split, we have to checkout all except the recording files.